### PR TITLE
Added compare before update

### PR DESCRIPTION
**Pull request by rschouten97**

### CODE REVIEW

**Comment by Rick-Jongbloed** [comment]
Please change mailboxes to groups (line 25)

File: Sync-AD-Groups-To-Products.ps1
Comment (Rick-Jongbloed) [inline Sync-AD-Groups-To-Products.ps1:1]: Line 539-542 should be

```
	if (-not[String]::IsNullOrEmpty($ADGroupsFilter)) {
		$adQuerySplatParams = @{
			Filter     = $ADGroupsFilter
			Properties = $properties
		}
	} else {
		$adQuerySplatParams = @{
			Properties = $properties
		}
	}
```

Comment (Rick-Jongbloed) [inline Sync-AD-Groups-To-Products.ps1:1]: Line 25: In in comment: rename mailboxes to groups 
Also line 25: Possibly change the comment to to "If all groups need to be queried,  the filter should be set to "*"

File: README.md
Comment (Rick-Jongbloed) [inline README.md:1]: line: | $ADGroupsFilter | String value of filter of which AD groups to include | Optional, when no filter is provided ($ADGroupsFilter = "*"), all groups will be queried |

Possibly change this to "If all groups need to be queried,  the filter should be set to "*"

**Review by Rick-Jongbloed — CHANGES_REQUESTED** [review]
See comments

File: Sync-AD-Groups-To-Products.ps1
Comment (rschouten97) [reply Sync-AD-Groups-To-Products.ps1:1]: Corrected.

**Pull request merged**